### PR TITLE
Fix replication for 5.6 as well

### PR DESCRIPTION
**Pull request by mnagy**
Start master properly, initializing correct configuration files.
Fix replication template as well to point to new image namespace.

OpenShift extended tests for mysql replication should work after this is picked-up.

@bparees @mfojtik @hhorak @soltysh @pmorie @smarterclayton 

Big kudos to @hhorak who found the main bug.

### CODE REVIEW

File: 5.6/root/usr/bin/run-mysqld-master [hunk 1]
```diff
@@ -5,16 +5,11 @@
...
-  exec /usr/bin/run-mysqld "$@"
-fi
-
 [ -f ${CONTAINER_SCRIPTS_PATH}/validate_replication_variables.sh ] && . ${CONTAINER_SCRIPTS_PATH}/validate_replication_variables.sh
 [ -f ${CONTAINER_SCRIPTS_PATH}/validate_variables.sh ] && . ${CONTAINER_SCRIPTS_PATH}/validate_variables.sh
 
-# Generate the unique 'server-id' for this master
-export MYSQL_SERVER_ID=$(server_id)
+# The 'server-id' for master needs to be constant
+export MYSQL_SERVER_ID=1
```
Comment (bparees) [inline 5.6/root/usr/bin/run-mysqld-master]: don't we need this change in the 5.5 scripts too?

Comment (bparees) [inline 5.6/root/usr/bin/run-mysqld-master]: nm, i'm blind.

**Comment by pmorie** [comment]
:+1:

**Comment by bparees** [comment]
failed [test]

**Comment by hhorak** [comment]
Looks fine, LGTM.

**Comment by bparees** [comment]
looks to me like the 5.6 rhel version keeps failing here:

2015-11-06 13:47:42 65 [Note] /opt/rh/rh-mysql56/root/usr/libexec/mysqld: ready for connections.
Version: '5.6.26-log'  socket: '/tmp/mysql.sock'  port: 0  MySQL Community Server (GPL)
/usr/share/container-scripts/mysql/common.sh: line 99:    65 Killed                  ${MYSQL_PREFIX}/libexec/mysqld --defaults-file=$MYSQL_DEFAULTS_FILE --skip-networking --socket=/tmp/mysql.sock "$@"
ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/tmp/mysql.sock' (111)

**Comment by hhorak** [comment]
Weird, it works fine on my machine (if that is the rhel7 image with the same test 5.6/test/run).

**Comment by bparees** [comment]
yeah not failing for me either... seeing other weirdness in the push job right now too.  will revisit this after getting that sorted out...wonder if dockerhub is having issues.

**Comment by mnagy** [comment]
Tests passed, merging.

**Pull request merged**